lib/index.js

```diff
@@ -47,78 +47,67 @@ module.exports = function loader (css, map) {
 
   validateOptions(require('./options.json'), options, 'PostCSS Loader')
 
-  const rc = {
-    path: path.dirname(file),
-    ctx: {
-      file: {
-        extname: path.extname(file),
-        dirname: path.dirname(file),
-        basename: path.basename(file)
-      },
-      options: {}
+  Promise.resolve().then(() => {
+    if (!options.config || (options.config && !options.config.path)) {
+      return parseOptions.call(this, options)
+    }
+
+    const rc = {
+      path: path.dirname(file),
+      ctx: {
+        file: {
+          extname: path.extname(file),
+          dirname: path.dirname(file),
+          basename: path.basename(file)
+        },
+        options: {}
+      }
     }
-  }
 
-  if (options.config) {
     if (options.config.path) {
       rc.path = path.resolve(options.config.path)
     }
 
     if (options.config.ctx) {
       rc.ctx.options = options.config.ctx
     }
-  }
-
-  const sourceMap = options.sourceMap
-
-  Promise.resolve().then(() => {
-    const length = Object.keys(options).length
-
-    // TODO
-    // Refactor
-    if (!options.config && !sourceMap && length) {
-      return parseOptions.call(this, options)
-    } else if (options.config && !sourceMap && length > 1) {
-      return parseOptions.call(this, options)
-    } else if (!options.config && sourceMap && length > 1) {
-      return parseOptions.call(this, options)
-    } else if (options.config && sourceMap && length > 2) {
-      return parseOptions.call(this, options)
-    }
 
     return postcssrc(rc.ctx, rc.path, { argv: false })
   }).then((config) => {
     if (!config) config = {}
 
     if (config.file) this.addDependency(config.file)
 
+    const sourceMap = options.sourceMap
+    delete options.sourceMap
```

Comment:
This caused problems on 'multiple' runs for some weird reason, but maybe because it was defined in the wrong place, not sure here tbh ¯\_(ツ)_/¯. It work for the first loader run and was then missing for all further runs, which lead to `const sourceMap` being always `undefined`. Besides that `delete options.sourceMap` => `options.sourceMap = null` (perf) ?